migrate matrix_exp to opInfo tests

bdhirsh opened the pull request:
Migrating `matrix_exp` to OpInfo testing.

It doesn't have an `out` variant, to I explicitly skipped `out` testing.

Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#55533 migrate matrix_exp to opInfo tests**

Differential Revision: [D27628966](https://our.internmc.facebook.com/intern/diff/D27628966)

bdhirsh committed:
migrate matrix_exp to opInfo tests

[ghstack-poisoned]

mruberry commented on torch/testing/_internal/common_methods_invocations.py:
```diff
@@ -1991,6 +1991,13 @@ def sample_inputs_masked_select(op_info, device, dtype, requires_grad, **kwargs)
 
+def sample_inputs_matrix_exp(op_info, device, dtype, requires_grad, **kwargs):
+    samples = (
+        SampleInput(make_tensor((S, S), device, dtype, requires_grad=requires_grad)),
```
Nice

mruberry commented on torch/testing/_internal/common_methods_invocations.py:
```diff
@@ -3213,6 +3220,10 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
            dtypesIfCUDA=all_types_and_complex_and(torch.bool, torch.half, torch.bfloat16),
            sample_inputs_func=sample_inputs_masked_select),
+    OpInfo('matrix_exp',
+           dtypes=floating_and_complex_types_and(torch.bfloat16),
```
From the failing tests it looks like this will need 

```
dtypesIfCPU=floating_and_complex_types_and(torch.bfloat16),
dtypesIfCUDA=floating_and_complex_types_and(torch.float16),
```

and to not set `dtypes` (just let it inherit the default)

mruberry reviewed (approved):
Cool! Thanks for participating in the OpInfo porting hackathon @bdhirsh. There are a few failing tests that should be fixed by updating the dtypes (see recommendation inline).

If that doesn't resolve the issue just ping me, otherwise this looks good!

bdhirsh committed:
Update on "migrate matrix_exp to opInfo tests"

Migrating `matrix_exp` to OpInfo testing.

It doesn't have an `out` variant, to I explicitly skipped `out` testing.


Differential Revision: [D27628966](https://our.internmc.facebook.com/intern/diff/D27628966)

[ghstack-poisoned]

mruberry commented on torch/testing/_internal/common_methods_invocations.py:
```diff
@@ -3213,6 +3220,11 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
            sample_inputs_func=sample_inputs_masked_select),
+    OpInfo('matrix_exp',
+           dtypesIfCPU=floating_and_complex_types_and(torch.bfloat16),
+           dtypesIfCUDA=floating_and_complex_types_and(torch.bfloat16),
```
CUDA needs `torch.float16`, not `torch.Bfloat16`, notice the missing 'b'

bdhirsh replied:
got it, thanks!

bdhirsh committed:
Update on "migrate matrix_exp to opInfo tests"

Migrating `matrix_exp` to OpInfo testing.

It doesn't have an `out` variant, to I explicitly skipped `out` testing.


Differential Revision: [D27628966](https://our.internmc.facebook.com/intern/diff/D27628966)

[ghstack-poisoned]

bdhirsh committed:
Update on "migrate matrix_exp to opInfo tests"

Migrating `matrix_exp` to OpInfo testing.

It doesn't have an `out` variant, to I explicitly skipped `out` testing.


Differential Revision: [D27628966](https://our.internmc.facebook.com/intern/diff/D27628966)

[ghstack-poisoned]

mruberry commented:
ROCm failure is unrelated. This is good to go!

bdhirsh committed:
Update on "migrate matrix_exp to opInfo tests"

Migrating `matrix_exp` to OpInfo testing.

It doesn't have an `out` variant, to I explicitly skipped `out` testing.


Differential Revision: [D27628966](https://our.internmc.facebook.com/intern/diff/D27628966)

[ghstack-poisoned]

bdhirsh committed:
Update on "migrate matrix_exp to opInfo tests"

Migrating `matrix_exp` to OpInfo testing.

It doesn't have an `out` variant, to I explicitly skipped `out` testing.


Differential Revision: [D27628966](https://our.internmc.facebook.com/intern/diff/D27628966)

[ghstack-poisoned]
